### PR TITLE
feat: Added clean data

### PR DESCRIPTION
**Pull request by SolMendiola**
# :star: Feature

---

#### :pencil2: Description:

Added clean data

---

### CODE REVIEW

File: lib/core/common/config.dart [hunk 1]
```diff
@@ -44,7 +44,7 @@ abstract class Config {
   static Uri xmartlabsInstagram =
       Uri.parse('https://www.instagram.com/xmartlabs/?hl=es');
   static Uri xmartlabsTwitter = Uri.parse('https://twitter.com/xmartlabs');
-
+  static String key = 'alreadyCleanData';
```
Comment (f7deleon) [inline lib/core/common/config.dart:47]: maybe a more descriptive key like `alreadyCleanDataKey`

File: lib/main.dart [hunk 2]
```diff
@@ -36,6 +38,16 @@ Future<void> main() async {
   );
 }
 
+Future<void> clearSecureStorageOnReinstall() async {
+  final SharedPreferences prefs = await SharedPreferences.getInstance();
+
+  if (!(prefs.getBool(Config.key) != null)) {
```
Comment (f7deleon) [inline lib/main.dart:44]: this checks for the key to be nullable i think that are two separate conditions or check directly the value you are expecting like `false`

File: lib/core/common/config.dart (hunk 1 above)
Comment (mirland) [reply lib/core/common/config.dart:47]: it shouldnt be in the config right?

Comment (mirland) [reply lib/core/common/config.dart:47]: It shouldn't be in the config right?

File: lib/main.dart (hunk 2 above)
Comment (mirland) [reply lib/main.dart:44]: why not just `!prefs.getBool(Config.key) == true`?

File: lib/main.dart [hunk 3]
```diff
@@ -36,6 +38,16 @@ Future<void> main() async {
   );
 }
 
+Future<void> clearSecureStorageOnReinstall() async {
+  final SharedPreferences prefs = await SharedPreferences.getInstance();
```
Comment (mirland) [inline lib/main.dart:42]: I'd use a provider, lets discuss it tomorrow, but we can follow [this pr](https://github.com/xmartlabs/flutter-template/pull/124)

File: pubspec.yaml [hunk 4]
```diff
@@ -54,6 +54,7 @@ dependencies:
   styled_text: 6.0.0
   url_launcher: 6.1.7
   webview_flutter: 4.0.1
+  shared_preferences: ^2.0.16
```
Comment (mirland) [inline pubspec.yaml:57]: alpha and hat

File: lib/main.dart [hunk 5]
```diff
@@ -36,6 +38,16 @@ Future<void> main() async {
...
+
+  if (!(prefs.getBool(Config.key) != null)) {
+    const FlutterSecureStorage storage = FlutterSecureStorage();
```
Comment (mirland) [inline lib/main.dart:45]: you are not injecting it

**Review by mirland — APPROVED** [review]

**Review by mirland — CHANGES_REQUESTED** [review]
Let check the comments before merge it

File: lib/core/common/config.dart [hunk 6]
```diff
@@ -44,7 +44,6 @@ abstract class Config {
   static Uri xmartlabsInstagram =
       Uri.parse('https://www.instagram.com/xmartlabs/?hl=es');
   static Uri xmartlabsTwitter = Uri.parse('https://twitter.com/xmartlabs');
-
   static const debugMode = kDebugMode;
```
Comment (mirland) [inline lib/core/common/config.dart:47]: I'd keep this line
```suggestion

  static const debugMode = kDebugMode;
```

File: lib/core/di/di_utils_module.dart [hunk 7]
```diff
@@ -11,10 +15,30 @@ class UtilsDiModule {
...
 
-extension _GetItUseCaseDiModuleExtensions on GetIt {
+extension _GetItUtilsDiModuleExtensions on GetIt {
   void _setupModule() {
     registerSingleton(AppRouter());
   }
+
+  void _setupProviders() {
+    registerSingletonAsync(
+      () => $FloorAppDatabase.databaseBuilder('database.db').build(),
```
Comment (mirland) [inline lib/core/di/di_utils_module.dart:32]: we can maybe add the DB name in the config file

**Review by mirland — APPROVED** [review]

**Pull request merged**